cmd/api/main.go

```diff
@@ -186,7 +193,7 @@ func main() {
 	ctx := context.Background()
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	mux := runtime.NewServeMux()
+	mux := runtime.NewServeMux(runtime.WithIncomingHeaderMatcher(impersonation.HeaderMatcher))
```

Comment:
Should this be added if the impersonation check is disabled?

Reply:
The default header matcher is implicit in `runtime.NewServeMux()`. This custom function is added so that the requester can send impersonation header with name `Impersonate-User` instead of GRPC gateway header format, `Grpc-Metadata-Impersonate-User`.

But I'll still put in this in a condition.

Reply:
https://github.com/tektoncd/results/blob/06adf689eb14324be872c43787d6d68861df5af9/cmd/api/main.go#L109-L112